### PR TITLE
Added cusoverDnZhegvdx to test_driver

### PR DESCRIPTION
**Pull request by mimaric**
CUDA 10.1 added `cusolverDnZhegvdx` and `cusolverDnDsygvdx` to cuSOLVER. This PR enables automatic switching between `cusolverDnZhegvdx`/`cusolverDnDsygvdx` and `cusolverDnZhegvd`/`cusolverDnDsygvd`.

In addition it removes custom implementation of `cusolverDn` module. That module is now provided by PGI.

### CODE REVIEW

File: test_driver/test_dsygvdx.F90 [hunk 1]
```diff
@@ -214,7 +229,11 @@ program main
   w2_d = 0
   ts = wallclock()
   call nvtxStartRange("cuSOLVER",5)
+#ifdef HAVE_CUSOLVERDNDSYGVDX
+  istat = cusolverDnDsygvdx(h, CUSOLVER_EIG_TYPE_1, CUSOLVER_EIG_MODE_VECTOR, CUSOLVER_EIG_RANGE_I, CUBLAS_FILL_MODE_UPPER, N, A2_d, lda, B2_d, lda, 0.D0, 0.D0, il, iu, h_meig, w2_d, work_d, lwork_d, devInfo_d)
+#else
   istat = cusolverDnDsygvd(h, CUSOLVER_EIG_TYPE_1, CUSOLVER_EIG_MODE_VECTOR, CUBLAS_FILL_MODE_UPPER, N, A2_d, lda, B2_d, lda, w2_d, work_d, lwork_d, devInfo_d)
+#endif
   call nvtxEndRange
   te = wallclock()
```
Comment (romerojosh) [inline test_driver/test_dsygvdx.F90:240]: Can you update the terminal output string after this runs to print `Time for cusolverDnDsygvdx` if the new API is used?

File: test_driver/test_zhegvdx.F90 [hunk 2]
```diff
@@ -218,7 +233,11 @@ program main
   w2_d = 0
   ts = wallclock()
   call nvtxStartRange("cuSOLVER",5)
+#ifdef HAVE_CUSOLVERDNZHEGVDX
+  istat = cusolverDnZhegvdx(h, CUSOLVER_EIG_TYPE_1, CUSOLVER_EIG_MODE_VECTOR, CUSOLVER_EIG_RANGE_I, CUBLAS_FILL_MODE_UPPER, N, A2_d, lda, B2_d, lda, 0.D0, 0.D0, il, iu, h_meig, w2_d, work_d, lwork_d, devInfo_d)
+#else
   istat = cusolverDnZhegvd(h, CUSOLVER_EIG_TYPE_1, CUSOLVER_EIG_MODE_VECTOR, CUBLAS_FILL_MODE_UPPER, N, A2_d, lda, B2_d, lda, w2_d, work_d, lwork_d, devInfo_d)
+#endif
   call nvtxEndRange
   te = wallclock()
```
Comment (romerojosh) [inline test_driver/test_zhegvdx.F90:244]: Similarly, can you update the terminal output string after this runs to print `Time for cusolverDnZhegvd` if the new API is used?

**Comment by romerojosh** [comment]
Looks good to me. Left a couple of small comments. Additionally, since we are removing the custom `cusolverDn` module, would you mind updating the minimum PGI compiler version in the building section of the `README.md` to 18.10? Thanks!

File: test_driver/test_dsygvdx.F90 (hunk 1 above)
Comment (mimaric) [reply test_driver/test_dsygvdx.F90:240]: done

File: test_driver/test_zhegvdx.F90 (hunk 2 above)
Comment (mimaric) [reply test_driver/test_zhegvdx.F90:244]: done

**Comment by romerojosh** [comment]
Thanks Milos! Merging.

**Pull request merged**